gunrock/app/bfs/bfs_kernel.cuh

```diff
@@ -0,0 +1,392 @@
+// ----------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------
+
+/**
+ * @file
+ * bfs_kernel.cuh
+ *
+ * @brief BFS GPU kernels
+ */
+
+#pragma once
+
+namespace gunrock {
+namespace app {
+namespace bfs {
+
+template <typename ProblemT, int LOG_THREADS>
+__global__ void From_Unvisited_Queue_IDEM(
```

Comment:
I understand that these kernels are called within the backwards portion of the DOBFS, but it is not apparent from the first look on what they actually do. Can you please write a small brief for each kernel in this file?